### PR TITLE
[Sehun Kim] EX1 解答 [python]

### PR DESCRIPTION
**Pull request by KimSehun725**
# WHAT
第１回B4輪講の演習課題
　1. 音声を収録してスペクトログラムを表示しよう
　2. スペクトログラムを逆変換して時間信号に戻そう

# RESULT
result

# WIP

### CODE REVIEW

File: ex1/sehunkim/ex01_Kim.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import numpy as np
+import librosa
+import librosa.display
+from matplotlib import pyplot as plt
+from scipy.fft import fft, ifft
+from scipy import signal
+
+#sampling rate = 44100Hz
```
Comment (Secondtonumb) [inline ex1/sehunkim/ex01_Kim.py:8]: Use [argparse](https://docs.python.org/ja/3/library/argparse.html) then you can set parameters while running the code rathen than changing variables everytime.
This is very important when you are writing huge code.

File: ex1/sehunkim/ex01_Kim.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+frame_size = 300                                        # size of one frame of stft
+overlap = 70                                            # number of samples to overlap
+window_func = signal.hamming(frame_size)                # variable for window function
+n_of_frame = len(wave_array) // (frame_size - overlap)  # calculating the number of frames
+p_wave_array = np.pad(wave_array,(0,overlap))           # zero padding
+time = wave_array / sr                                  # calculating total time
+
+def stft(data_array, frame_size, overlap):
+    result = np.array([])
+    for frame_no in range (n_of_frame) :
```
Comment (Secondtonumb) [inline ex1/sehunkim/ex01_Kim.py:19]: too much space

File: ex1/sehunkim/ex01_Kim.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        frame = frame * window_func                     # apply window function
+        result = np.append(result, fft(frame))          # append the result of fft
+    result = result.reshape(n_of_frame,frame_size)
+    return result
+
+def istft (data_array, frame_size, overlap):
+    result = np.array([])
+    for frame_no in range (n_of_frame) :
+        result = np.append(result, ifft(data_array[frame_no])[:frame_size - overlap])
+    return result
```
Comment (Secondtonumb) [inline ex1/sehunkim/ex01_Kim.py:32]: Frame shift should also be done here.
```
result[start: start+win_frame] = ...
start += win_frame -overlap
result[start: start+win_frame] = ...
```

File: ex1/sehunkim/ex01_Kim.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+# plot re-synthesized waveform
+p_istft_result = np.pad(istft_result,(0, len(t) - len(istft_result))) # zero padding
+plt.subplot(3,1,3)
+plt.plot(t, p_istft_result)
+plt.title('Re-synthesized Signal')
+plt.xlabel('Time[s]')
+plt.ylabel('Magnitude')
+plt.show()
+
```
Comment (Secondtonumb) [inline ex1/sehunkim/ex01_Kim.py:67]: with`plt.ylim` you can set limitation for each axis and align subplot up.

**Review by Secondtonumb — CHANGES_REQUESTED** [review]
It's a nice way to pad the whole `wav_array ` before windowing.Good Job.
And I can tell that you've almostly under how stft/istft works and most of the part is correct(Such as you kept y-axis label in linear style except of changing them into 'log' which is wrong), just slightly debug and code format changing is needed.
頑張ってください！

**Review by Secondtonumb — COMMENTED** [review]
Add you new synthesized plot and I'll approve it!

**Review by yoshikawa-tatsuya — APPROVED** [review]
よく書けていると思います。argparseを使えるとなお良いと思います。

**Pull request merged**